blog/articles/distributed-tracing.md

```diff
@@ -0,0 +1,488 @@
+EOF
+```
+
+We'll update our heartbeats service to start sending heartbeats here instead:
+
+```
+kubectl apply -f - <<EOF
+apiVersion: sources.knative.dev/v1
+kind: ContainerSource
+metadata:
+  name: heartbeats
+spec:
+  template:
+    spec:
+      containers:
+        - image: benmoss/heartbeats:latest
+          name: heartbeats
+          args:
+            - --period=1
+          env:
+            - name: POD_NAME
+              value: "heartbeats"
+            - name: POD_NAMESPACE
+              value: "default"
+            - name: K_CONFIG_TRACING
+              value: '{"backend":"zipkin","debug":"true","zipkin-endpoint":"http://otel-collector.observability:9411/api/v2/spans"}'
+  sink:
+    ref:
+      apiVersion: serving.knative.dev/v1
+      kind: Service
+      name: event-display
+EOF
+```
+
+Once these services are deployed we can check back over at our Jaeger dashboard and we should see some more a more interesting trace:
+
+![Screenshot of the Jaeger UI of a traces](/blog/articles/images/tracing-1.png)
+
+On the “System Architecture” tab of Jaeger we can also see a nice graph of our
+topology, including one component you may or may not know about, the
+[activator](https://knative.dev/docs/serving/knative-kubernetes-services/#service-activator):
+
+![Screenshot of a Jaeger architecture diagram](/blog/articles/images/tracing-2.png)
+
+This is a component that Knative Serving adds to the network path of Knative
+Service to buffer requests when our service isn’t ready to handle them as well
+as reporting request metrics to the autoscaler. You can also see that it adds a
+tiny penalty, around 2ms on my cluster. It’s possible to [configure
+Knative](https://knative.dev/docs/serving/load-balancing/target-burst-capacity/)
+so that the activator stays out of the path under different scenarios, but
+that’s a topic for another blog post :).
+
+
+## Getting fancy
+
+Let’s make our topology a little more interesting by adding in some Knative
+bells and whistles. First, let’s start sending our messages through a
+[Broker](https://knative.dev/docs/eventing/broker/) and
+[Trigger](https://knative.dev/docs/eventing/broker/triggers/) instead of
+directly from our heartbeats service. We’ll create a broker and trigger that
+forward all messages to the event-display service, and reconfigure our heartbeat
+service to point to the broker:
+
+
+```
+kubectl apply -f - <<EOF
+apiVersion: eventing.knative.dev/v1
+kind: Trigger
+metadata:
+  name: heartbeat-to-eventdisplay
+spec:
+  broker: default
+  subscriber:
+    ref:
+      apiVersion: serving.knative.dev/v1
+      kind: Service
+      name: event-display
+---
+apiVersion: eventing.knative.dev/v1
+kind: Broker
+metadata:
+  name: default
+---
+apiVersion: sources.knative.dev/v1
+kind: ContainerSource
+metadata:
+  name: heartbeats
+spec:
+  template:
+    spec:
+      containers:
+        - image: benmoss/heartbeats:latest
```

Comment:
Can we use the upstream image instead of the personal image?
```suggestion
        - image: gcr.io/knative-releases/knative.dev/eventing/cmd/heartbeats
```